File: Cli_Core_C/Cli_Module.c
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

#include "Cli_Module.h"

static int Execute(struct Cli_Module *module, struct Cli_Cmd *cmd, struct Level_Description_Array *Levels, int is_debug) {
    return 0; /*Failed*/
}

void Cli_Module_Init_Base(struct Cli_Module *module, char *name) {
    module->Cmd_Head = NULL;
    module->Cmd_Tail = NULL;
    module->Module_Next = NULL;
    if (!name) {
        int len = CLI_MODULE_NAME_SIZE_DEF - 1;
        strncpy(module->Name, "Base", len);
        module->Name[len] = '\0';
    } else {
        int len = CLI_MODULE_NAME_SIZE_DEF - 1;
        strncpy(module->Name, name, len);
        module->Name[len] = '\0';
    }

    module->Execute = Execute;
}

struct Cli_Module Cli_Module_Init(char *name) {
    struct Cli_Module module;
    Cli_Module_Init_Base(&module, name);
    return module;
}

void Cli_Module_Add(struct Cli_Module *module_parent, struct Cli_Module *module) {
    module_parent->Module_Next = module;
    module->Module_Next = NULL;
}

void Cli_Module_Add_Cmd(struct Cli_Module *module, struct Cli_Cmd *cmd) {
    if (!module->Cmd_Head) {
        module->Cmd_Head = cmd;
        module->Cmd_Tail = cmd;
    } else {
        module->Cmd_Tail->Cmd_Next = cmd;
        module->Cmd_Tail = cmd;
    }
}

File: Cli_Core_JavaFX/src/cli_core_javafx_test/Cli_Module.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package cli_core_javafx_test;

import java.util.ArrayList;
import java.util.List;
import java.util.Map;

/**
 *
 * @author mike
 */
public class Cli_Module {

    protected String Name;

    public List<Cli_Cmd> Module_Cmd_List;

    public int Cmd_ID_Count_Get() {
        return 0;
    }

    public Cli_Module(String name) {
        Name = name;
        Module_Cmd_List = new ArrayList<>();
    }

    public String Name_Get() {
        return Name;
    }

    public void Cmd_Add(Cli_Cmd cmd) {
        Module_Cmd_List.add(cmd);
    }

    public boolean Execute(Cli_Cmd cmd, List<Level_Description> Levels, boolean is_debug) {
        return false;
    }

    public int Cmd_Count_Get() {
        return Module_Cmd_List.size();
    }

    public Cli_Cmd Cmd_Get(int cmd_index) {
        if (cmd_index >= 0 && cmd_index < Module_Cmd_List.size()) {
            return Module_Cmd_List.get(cmd_index);
        }
        return null;
    }

    public void To_Map(Map<String, String> values_map) {
        // Nothing
    }

}
